plugins/explore/src/plugin.cpp

```diff
@@ -0,0 +1,235 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "explore/components.hpp"
+#include "explore/operator_args.hpp"
+#include "explore/printer_args.hpp"
+#include "explore/ui_state.hpp"
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/data.hpp>
+#include <tenzir/error.hpp>
+#include <tenzir/logger.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/table_slice.hpp>
+
+#include <ftxui/component/loop.hpp>
+#include <ftxui/component/screen_interactive.hpp>
+#include <ftxui/screen/screen.hpp>
+
+#include <thread>
+
+namespace tenzir::plugins::explore {
+
+namespace {
+
+/// Construct an FTXUI screen from the plugin configuration.
+auto make_interactive_screen(const operator_args& args)
+  -> ftxui::ScreenInteractive {
+  using namespace ftxui;
+  TENZIR_ASSERT((args.width && args.height) || (!args.width && !args.height));
+  TENZIR_ASSERT(!args.width
+                || (args.width->inner > 0 && args.height->inner > 0));
+  if (args.width && args.height)
+    return ScreenInteractive::FixedSize(args.width->inner, args.height->inner);
+  if (args.fullscreen)
+    return ScreenInteractive::Fullscreen();
+  if (args.fit)
+    return ScreenInteractive::FitComponent();
+  return ScreenInteractive::TerminalOutput();
+}
+
+class explore_operator final : public crtp_operator<explore_operator> {
+public:
+  explore_operator() = default;
+
+  explicit explore_operator(operator_args args) : args_{std::move(args)} {
+  }
+
+  auto name() const -> std::string override {
+    return "explore";
+  }
+
+  auto operator()(generator<table_slice> input) const
+    -> generator<std::monostate> {
```

Comment:
One small comment given a recent API change so that I don't forget about it when I eventually get to review this:

This function needs to also take the operator control plane as an argument, and then have something like this at the very start:

```cpp
if (not ctrl.has_terminal()) {
  diagnostic::error("…")
    .emit(ctrl.diagnostics());
  co_return;
}
```

Reply:
Fixed in 1894a1d6e7.